apps/aecore/src/aec_conductor_chain.erl

```diff
@@ -211,50 +214,27 @@ resolve_name(Type, Name, State) ->
 %%% Handle persistence
 
 persistence_store_block(Block, State) ->
-    try begin
-            %% Best effort persistence.
-            %% If the server is not there, ignore it.
-            #block{txs = Transactions} = Block,
-            aec_db:ensure_transaction(
-              fun() ->
-                      aec_db:write_txs(
-                        Transactions, block_hash(Block)),
-                      aec_db:write_block(Block),
-                      persist_chain(State)
-              end)
-        end
-    catch T:E ->
-            lager:error("Persistence server error: ~p:~p", [T, E]),
-            ok
-    end.
-
-persistence_store_header(Header, State) ->
-    try begin
-            %% Best effort persistence.
-            %% If the server is not there, ignore it.
-            aec_db:ensure_transaction(
-              fun() ->
-                      aec_db:write_header(Header),
-                      persist_chain(State)
-              end)
-        end
-    catch T:E ->
-            lager:error("Persistence server error: ~p:~p", [T, E]),
-            ok
-    end.
+    #block{txs = Transactions} = Block,
+    aec_db:ensure_transaction(
+      fun() ->
+              aec_db:write_txs(Transactions, block_hash(Block)),
+              persist_chain(State)
+      end).
 
 persist_chain(State) ->
-    aec_db:ensure_transaction(fun() -> do_persist_chain(State) end).
-
-do_persist_chain(State) ->
-    case aec_chain_state:top_header_hash(State) of
+    case aec_chain_state:get_genesis_hash(State) of
         undefined -> ok;
```

Comment:
I see that `init_state` guarantees the genesis hash to be present. I see this `persist_chain`function is called when inserting a block - *including genesis* - or header.

Do I understand correctly that, maybe by making the `insert_block` function call in `init_state` more specific, this function `persist_chain` can be made to not conditionally not-persist chain? I find that a function `persist_chain` that may *not* do any side effect for a case that I understand shall not happen may be improved in readability.